src/main/scala/com/microsoft/hyperspace/index/rules/FilterIndexRule.scala

```diff
@@ -66,27 +70,52 @@ object FilterIndexRule extends Rule[LogicalPlan] with Logging {
             .flatMap(_.toSeq)
           val filterColumnNames = condition.references.map(_.name).toSeq
 
-          replaceWithIndexIfPlanCovered(
-            project,
+          val transformedPlan = replaceWithIndexIfPlanCovered(
+            filter,
             projectColumnNames,
             filterColumnNames,
             logicalRelation,
             fsRelation,
             location)
+          project.copy(child = transformedPlan)
         } catch {
           case e: Exception =>
             logWarning("Non fatal exception in running filter index rule: " + e.getMessage)
             project
         }
+
+      case filter @ Filter(
+            condition: Expression,
+            logicalRelation @ LogicalRelation(
+              fsRelation @ HadoopFsRelation(location, _, _, _, _, _),
+              _,
+              _,
+              _)) =>
+        try {
```

Comment:
Did you try refactoring this as discussed offline?

Reply:
Using extractor pattern is done.